src/main/java/com/powsybl/openloadflow/network/impl/PropagatedContingency.java

```diff
@@ -320,27 +344,28 @@ private static Identifiable<?> getIdentifiable(Network network, ContingencyEleme
     }
 
     public boolean hasNoImpact() {
-        return branchIdsToOpen.isEmpty() && hvdcIdsToOpen.isEmpty() && generatorIdsToLose.isEmpty()
+        return branchIdsToOpen.isEmpty()
+                && hvdcIdsToOpen.isEmpty() && generatorIdsToLose.isEmpty()
                 && loadIdsToLoose.isEmpty() && shuntIdsToShift.isEmpty() && busIdsToLose.isEmpty();
     }
 
     public Optional<LfContingency> toLfContingency(LfNetwork network) {
-        // update connectivity with triggered branches of this network
+        // update connectivity with triggered lostBranches of this network
         GraphConnectivity<LfBus, LfBranch> connectivity = network.getConnectivity();
         connectivity.startTemporaryChanges();
 
-        List<LfBranch> branchesToOpen = branchIdsToOpen.stream()
+        List<LfBranch> branchesToOpen = branchIdsToOpen.keySet().stream()
                 .map(network::getBranchById)
                 .filter(Objects::nonNull) // could be in another component
                 .collect(Collectors.toList());
 
-        // we add the branches connected to buses to lose.
+        // we add the lostBranches connected to lostBuses to lose.
         busIdsToLose.stream().map(network::getBusById)
```

Comment:
`BusIdsToLose` are the list of lost buses when the network is in bus/breaker topology. Connected to a lost bus, we can have:
- Line connected at bus side only: in that case, the branch is disabled.
- Line connected at both side: in that case, we have to disconnect only the bus side terminal. 
- Line connected only at the other side. I though we don't get it when we do a bus.getBranches() but in case of an automation system that has open the bus side, could it be in the future?